### PR TITLE
Add server settings system keys data source

### PR DESCRIPTION
**Pull request by PingDavidR**
- Closes Brass 1258

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@
 * **New Data Source:** `protocol_metadata_lifetime_settings datasource` ([#100](https://github.com/pingidentity/terraform-provider-pingfederate/pull/100))
 * **New Data Source:** `server_settings_general_settings datasource` ([#101](https://github.com/pingidentity/terraform-provider-pingfederate/pull/101))
 * **New Data Source:** `server_settings_log_settings datasource` ([#104](https://github.com/pingidentity/terraform-provider-pingfederate/pull/104))
+* **New Data Source:** `server_settings_general_settings datasource` ([#105](https://github.com/pingidentity/terraform-provider-pingfederate/pull/105))
+* **New Data Source:** `server_settings_system_keys datasource` ([#112](https://github.com/pingidentity/terraform-provider-pingfederate/pull/112))
```
Comment (henryrecker-pingidentity) [inline CHANGELOG.md:12]: You can remove "datasource" from these to match the format elsewhere in the file

File: internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readServerSettingsSystemKeysDataSource(ctx context.Context, r *client.SystemKeys, state *serverSettingsSystemKeysDataSourceModel, existingId *string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	state.Id = id.GenerateUUIDToState(existingId)
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go:138]: This is a singleton data source, so we can use a hardcoded id here. There's no way to track an existing id with data sources.

File: internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	Id       types.String `tfsdk:"id"`
+	Current  types.Object `tfsdk:"current"`
+	Previous types.Object `tfsdk:"previous"`
+	Pending  types.Object `tfsdk:"pending"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *serverSettingsSystemKeysDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes the settings for server system keys.",
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go:45]: This isn't really describing settings, it just returns the system keys themselves.

File: internal/resource/config/serversettings/systemkeys/server_settings_system_keys_resource.go [hunk 4]
```diff
@@ -220,23 +220,26 @@ func readServerSettingsSystemKeysResponse(ctx context.Context, r *client.SystemK
 		"encrypted_key_data": types.StringValue(currentAttrs.GetEncryptedKeyData()),
 		"key_data":           types.StringValue(currentAttrs.GetKeyData()),
 	}
-	currentAttrsObjVal := internaltypes.MaptoObjValue(systemKeyAttrTypes, currentAttrVals, &diags)
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/serversettings/systemkeys/server_settings_system_keys_resource.go:223]: Was there an issue with this method?

**Review by henryrecker-pingidentity — CHANGES_REQUESTED** [review]

File: internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go (hunk 2 above)
Comment (PingDavidR) [reply internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go:138]: ah, copy/paste for a fix.  good catch

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**